org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -1584,6 +1584,9 @@ public static IMethod resolveMethod(CompletionProposal proposal, IJavaProject ja
 		char[] declarationSignature = proposal.getDeclarationSignature();
 		String typeName = SignatureUtil.stripSignatureToFQN(String.valueOf(declarationSignature));
 		IType type = javaProject.findType(typeName);
+		if (type == null) {
+			type = javaProject.findType(typeName, new NullProgressMonitor());

```

Comment:
See javadoc for IType findType(String fullyQualifiedName, IProgressMonitor progressMonitor) throws JavaModelException;

> Same functionality as findType(String) but also looks for secondary types if the given name does not match a compilation unit name.

You can also try the org.eclipse.jdt.ls.core.internal.handlers.SignatureHelpHandlerTest.testSignatureHelpMethod() test without it.

Reply:
Since findType might be a long-running operation, how about adding a new param `IProgressMonitor progressMonitor` to this method as a flag to denote whether the secondary type will be checked? This also makes the signature similar to the `IJavaProject#findType()`.

Reply:
So `resolveMethod` gets the additional parameter indicating whether the secondary type is checked. Sounds reasonable. I'd probably limit the secondary check to just this call for now since this is the only place we've been shown to need it.

@snjeza , let's make this change (assuming there are no problems with it) and then we can merge.

Reply:
Fixed.